### DISCUSSION TITLE
TechniSat support

### DISCUSSION THREAD

**Discussion by thenicnic** [PredatH0r/ChanSort#357] [Ideas]
Super Software, vielen Dank!

Gab es bereits Überlegungen, Exports von TechniSat-Receivern und -TVs zu unterstützen?
Der Export geschieht im Binärformat "CDP", bisher bin ich daraus aber noch nicht schlau geworden. Eine Uralt-Beta von 2011 von einer TechniSat-Software kursiert im Netz, basierend auf Java 6, die auch noch einwandfrei funktioniert.
In der Hoffnung auf Infos zum Parser könnte man den Code dekompilieren, das ist rechtlich natürlich schwierig.
Da es auch keinen Header in den CBP-Dateien gibt, sind sie vermutlich verschlüsselt.
Wenn Integration von TechniSat-Formaten in ChanSort gewünscht ist, möchte ich sehr gerne unterstützen.
Der Editor heißt "TechniPort Beta", ein Beispiel eines Receiver-Exports gibts hier: https://github.com/thenicnic/RandomUploads/blob/main/database.cdp

Danke!

**Comment by PredatH0r** [comment]
Danke für den Hinweis.
Die Datei sieht tatsächlich verschlüsselt aus. Ich werde mir gerne die Datei bzw. "TechniPort Plus Beta" einmal näher ansehen, sobald ich dafür wieder Zeit habe. In den nächsten 1-2 Monaten sieht es aber eher schlecht aus, da bin ich mit Haussanierung beschäftigt.

**Comment by PredatH0r** [comment]
Ich hätte jetzt versucht, eine database.cdp Datei mit "TechniPort Plus 0.9.5.4 Beta" zu öffnen, habe aber keine Möglichkeit gefunden.
USB-Stick "vorbereiten" und die Datei hinaufzukopieren funktioniert nicht und "Gespeicherte Favoritenliste öffnen" akzeptiert auch nur ".tvtppc" Dateien. Das Programm ist auch mit C++ und Qt entwickelt worden und keine Java-Anwendung, somit kann man sie auch nicht dekompilieren.

**Reply by PredatH0r** [reply]
Habe jetzt gesehen, dass "TechniPort Plus" etwas anders ist als "TechniPort" und auch einen Download gefunden.
Damit funktioniert es!

**Comment by thenicnic** [comment]
Vielen Dank!
"Eventuell" hätte ich die Datenbank bereits entschlüsselt, weiß aber nicht inwiefern die Diskussion darüber hier gewünscht ist.
Wollte kommende Woche mal bei TechniSat zwecks Kooperation nachfragen. 
Darf ich dich per E-Mail kontaktieren?
Viel Erfolg bei der Sanierung - auf dass du den Sommer trotzdem genießen kannst!

**Comment by PredatH0r** [comment]
Klar kannst du mich per e-mail kontaktieren. horst at beham dot biz.
Hab mittlerweile auch schon die ".exe" durch einen Java-Decompiler geschickt und die diversen Files im "tools" Ordner gefunden :)

Muss jetzt leider weiterspachteln, bevor mir das Zeug im Kübel fest wird.

**Comment by MarGeb1** [comment]
Hallo,

hat sich denn bezüglich der Unterstützung für Technisat etwas ergeben?

Markus
